src/pysdmx/util/__init__.py

```diff
@@ -25,6 +25,11 @@ class Reference(Struct, frozen=True):
     id: str
     version: str
 
+    @property
+    def full_id(self) -> str:
```

Comment:
I would be a bit careful with introducing new ways to reference artefacts. SDMX has 2:
- One where we need to use the artefact type, its ID, agency and version;
- One where we can use the SDMX URN.

Do we need a 3rd one?

Reply:
It is only a convenience method to extract the "full_id", i.e, the combination of agency_id, id and version (don´t know if there is a formal name for it). We could add it to utils

Reply:
That sounds OK to me. It would be nice to find a good name for it. For me, there is the ID and the URN. I also like the "relevant part of the URN", i.e. Type=Agency:Id(Version), as it contains all information while being shorter than the URN. 

But a full ID without the type sounds a bit odd, without knowing how it's used. After all, Codelist=SDMX:FREQ(1.0) and ConceptScheme=SDMX:FREQ(1.0) would have the same `full_id` but would nevertheless be 2 different artefacts...

Reply:
Could we then, as Reference is defined also with sdmx_type, make the __str__ method as sdmxtype=AGENCY_ID:ID(VERSION)? This should cover all use cases.

Reply:
And only call it Reference, I cannot find a better way to define it.

Reply:
Yes, if you are happy with adding the type too, then, this is perfect, thanks :). This then is the "relevant part of the URN" I was mentioning previously.

We could return it via a `__str__` method, as you suggested, which is nice, I think 😊. Or we can have a property, but, instead of calling it `full_id`, I would call it something like `short_urn`. 

But I think I prefer your suggestion of using `__str__`

Reply:
Implementing it as __str__. On Reference, should we add a post_init check for the sdmx_type? What should be the values that sdmx_type may take? Is there any list of those values?

Reply:
Thanks, @javihern98, this is nice :)

Re. your question, we could add this. As you may remember, I defined such an enum in the Structure query builder. Once it's ready, I could add it to the model instead. But, for the time being, I guess it's OK like that?